tools/generators/pbxproject_prefix/src/Generator/CalculateProjectDir.swift

```diff
@@ -0,0 +1,14 @@
+import PBXProj
+
+extension Generator {
+    /// Calculates the `PBXProject.projectDir` string.
+    ///
+    /// - Parameters:
+    ///   - executionRoot: The absolute path to the Bazel execution root.
+    static func calculateProjectDir(executionRoot: String) -> String {
```

Comment:
See other comment on using the word "calculate", why not simply call it `projectDir`? No strong preference here btw, feel free to ignore!

Reply:
I'm open to better names. Is `projectDir(executionRoot:)` clear? I'm bad at function names, but if this is correct Swift style, I'm all for it!

Reply:
Since this function doesn't have any side effects, I think simply calling it `projectDir` is fine.

Reply:
+1